woodwork/table_accessor.py

```diff
@@ -72,6 +72,20 @@ def init(self,
         _validate_accessor_params(self._dataframe, index, make_index, time_index, logical_types, schema)
         if schema is not None:
             self._schema = schema
+            extra_params = []
+            if index is not None:
+                extra_params.append('index')
+            if make_index:
+                extra_params.append('make_index')
+            if time_index is not None:
+                extra_params.append('time_index')
+            if logical_types is not None:
+                extra_params.append('logical_types')
+            for key in kwargs:
+                extra_params.append(key)
+            if extra_params:
+                warnings.warn("A schema was provided and the following parameters were ignored: " + ", ".join(extra_params), Warning)
```

Comment:
We have started to implement custom warning classes to easily allow users to suppress specific warnings they don't care about. I think we should follow that pattern here. Maybe create a new warning class inside `exceptions.py`, maybe something like `ParameterIgnoredWarning`, and use that here instead of the base `Warning` class.